src/containers/SignUp/signUp.jsx

```diff
@@ -1,145 +1,179 @@
-import React from 'react';
+import React, { Component } from 'react';
 
 import { Form, Input, Icon, Button } from 'antd';
 import { Link } from 'react-router-dom';
 import PropTypes from 'prop-types';
 
 import Header from '../../components/Header';
-import { SIGN_IN } from '../../constants/routes';
+import { SIGN_IN, HOME } from '../../constants/routes';
 
 import FacebookButton from '../../components/FacebookButton';
 import GoogleButton from '../../components/GoogleButton';
+
+import { withFirebase } from '../Firebase';
 import './signUp.css';
 
-const SignUpForm = props => {
-  const {
-    form: { getFieldDecorator, validateFieldsAndScroll },
-    history: { goBack },
-  } = props;
+class SignUpForm extends Component {
+  state = {
+    errorMessage: '',
+  };
 
-  const handleSubmit = e => {
+  handleSubmit = e => {
+    const {
+      firebase,
+      form: { validateFieldsAndScroll },
+      history: { push },
+    } = this.props;
     e.preventDefault();
     validateFieldsAndScroll((err, values) => {
       if (!err) {
-        // eslint-disable-next-line no-console
-        console.log('Received values of form: ', values);
+        firebase
+          .doCreateUserWithEmailAndPassword(values.email, values.password)
+          .then(result =>
+            firebase
+              .user(result.user.uid)
+              .set(
+                {
+                  name: values.name,
+                  email: values.email,
+                  userID: result.user.uid,
```

Comment:
we need to add field for goal also , so add it like this : 
```suggestion
                  name: values.name,
                  email: values.email,
                  userID: result.user.uid,
                  goal: '',
                  
                              
```